### ISSUE TITLE
[BUG] idea to reduce duplicate code in algos.ipynb

### ISSUE THREAD

**Issue by lovettchris** [microsoft/archai#165]
**Describe the bug**

When going through the notebooks in the intended order, there is a lot of duplicate code in algos.ipynb, you could eliminate that duplicate code using this trick:

```python
import nbimporter
from search_space import MyModel, CNNSearchSpaceExt as CNNSearchSpace
```

**Comment by lovettchris** [comment]
But alas you have actually made some changes, can we sync the code so they are not different?

![image](https://user-images.githubusercontent.com/18707114/217414923-20e35b41-c347-4233-986e-2a92ec816f54.png)

**Comment by piero2c** [comment]
Fixed on #173
